### PR TITLE
Db cleanup

### PR DESCRIPTION
**Pull request by Rikmorn**
little cleanup on how you bootstrap the db, plus all this stuff should be dockerized so we can have consistent environments

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+FROM node:10-alpine as deps
+
+COPY package.json yarn.lock ./
+
+RUN yarn && \
```
Comment (robcalcroft) [inline Dockerfile:5]: Would just `yarn --modules-folder node_modules_prod` on its own work? I think `yarn` and `yarn install` do the same thing

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+FROM node:10-alpine as deps
```
Comment (robcalcroft) [inline Dockerfile:1]: Can we use LTS node here

File: README.md [hunk 3]
```diff
@@ -16,16 +16,30 @@ Add a `.env` file in the root of the project. You will need to add config for co
...
+
+```sh
+docker-compose up -d && yarn start:app
+```
+
 ### Scripts
 
 For a list of all scripts see the `scripts` field in the [package.json](package.json).
+
+### DB
```
Comment (robcalcroft) [inline README.md]: Can we use `Database` instead of `DB`, same with the body of this section

File: db/00-create-tables.sql [hunk 4]
```diff
@@ -0,0 +1,28 @@
+create table if not exists users(
```
Comment (robcalcroft) [inline db/00-create-tables.sql:1]: What is the numeric file naming for?

File: package.json [hunk 5]
```diff
@@ -26,17 +26,13 @@
...
-    "db:init": "run-s db:drop db:setup db:add-data",
-    "db:setup": "node scripts/setupDb",
     "format": "prettier --write \"**/*\"",
     "start": "run-p start:*",
     "start:api": "nodemon -q --watch src/api src/api/index.js",
     "start:app": "parcel src/app/index.html --open",
     "build:app": "parcel build src/app/index.html",
     "test": "run-p test:*",
-    "test:format": "prettier --check \"**/*\"",
+    "test:format": "prettier --check \"src/**/*\"",
```
Comment (robcalcroft) [inline package.json]: The was deliberately unspecific as it lets us format dot files like config

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+version: "2.2"
```
Comment (robcalcroft) [inline docker-compose.yaml:1]: In prod my plans are to use RDS not run our own postgres instance so would having the db dockerised like this provide us with any value?

**Review by robcalcroft — COMMENTED** [review]
Few comments about some decisions made here but looks cool, we can discuss how this fits into our serverless-as-possible approach for this app. Will let @benji6 do his review too as he wrote a lot of the db utils

File: Dockerfile (hunk 1 above)
Comment (Rikmorn) [reply Dockerfile:5]: that is creating a dependencies image for us to use on the next two steps. We need dev dependencies to run tests, and we only need production dependencies to run production. This approach allows us to cache both for use on demand and speed up our builds considerably.

File: Dockerfile (hunk 2 above)
Comment (Rikmorn) [reply Dockerfile:1]: 10 is LTS, the package.json is specifying v10 as well, can upgrade if needed

File: README.md (hunk 3 above)
Comment (Rikmorn) [reply README.md]: 👍

File: db/00-create-tables.sql (hunk 4 above)
Comment (Rikmorn) [reply db/00-create-tables.sql:1]: PG executes init scripts by alfabetical order. We need to make the tables first and then add the data, thus the numbers. This is a pretty standard approach.

File: package.json (hunk 5 above)
Comment (Rikmorn) [reply package.json]: will remove

File: docker-compose.yaml (hunk 6 above)
Comment (Rikmorn) [reply docker-compose.yaml:1]: absolutely, you can make your own clusters now and get an environment very similar to production. Plus this ensure we have a consistent environment at all time, especially for other contributors. It should never be required to install and configure extra software to run a project (this is how we get it runs of my computer excuses!)

**Comment by benji6** [comment]
I like the database stuff, a lot nicer than my ad hoc scripts and I was secretly running postgres in Docker so I'm fine with that. I'm not particularly bothered about developing with docker-compose given that this app is just universal node and I'm not sure we're even settled on docker in production yet, but happy to give this setup a go so LGTM pending Rob's comments

File: Dockerfile (hunk 1 above)
Comment (robcalcroft) [reply Dockerfile:5]: Sweet

File: db/00-create-tables.sql (hunk 4 above)
Comment (robcalcroft) [reply db/00-create-tables.sql:1]: Ahh gotcha, so do these get run by the postgres docker image on boot?

File: docker-compose.yaml (hunk 6 above)
Comment (robcalcroft) [reply docker-compose.yaml:1]: Nice! Gotcha

File: db/00-create-tables.sql (hunk 4 above)
Comment (Rikmorn) [reply db/00-create-tables.sql:1]: Correct, thus the volume on the compose file

File: Dockerfile (hunk 2 above)
Comment (robcalcroft) [reply Dockerfile:1]: My only thought was the using `node:lts` saves us one more line of config to change in the future

File: package.json (hunk 5 above)
Comment (robcalcroft) [reply package.json]: cool

File: Dockerfile (hunk 2 above)
Comment (Rikmorn) [reply Dockerfile:1]: Correct, but lts version changes over time, unless we remove the 
`"engines": {
    "node": "10"
  },`
from the package file then builds will fail once lts changes in the future, either remove the requirements from the package file or keep them consistent in both places

Comment (robcalcroft) [reply Dockerfile:1]: Yep you're right that makes a lot of sense

**Review by robcalcroft — APPROVED** [review]
LGTM 🙌

**Comment by Rikmorn** [comment]
👾

**Pull request merged**